holoviews/core/spaces.py

```diff
@@ -817,6 +817,22 @@ def _cache(self, key, val):
         self.data[key] = val
 
 
+    def map(self, map_fn, specs=None, clone=True):
+        """
+        Recursively replaces elements using a map function when the
+        specification applies. Extends regular map with functionality
+        to dynamically apply functions.
+        """
```

Comment:
Added one more test to check the clone=False condition, I generally prefer not adding multiple unit tests that just test end up testing the same code paths and Dimensioned.map is already tested.